### DISCUSSION TITLE
How to setup ViteJS with tailwindcss?

### DISCUSSION THREAD

**Discussion by itsbravestone** [StellaDocs/Reactwind#9] [Q&A] [answered]
Can anybody tell me how to setup ViteJS with tailwindcss?

**Comment by imarpanpatra** [comment, marked as answer]
Alright, I will tell you how to do so

**Step 1: Create Your Vite Project**
First, let’s get Vite going. Open your terminal and run:
`npm create vite@latest my-vite-app`

Pick a framework (React, Vue, or plain old Vanilla JS—whatever floats your boat), then head into your shiny new project:
`cd my-vite-app`

**Step 2: Install Tailwind CSS**

Time to bring in Tailwind CSS. Run this:
`npm install -D tailwindcss postcss autoprefixer`

And because life isn’t already full of enough config files, let’s add one more:
`npx tailwindcss init`

**Step 3: Tell Tailwind Where to Look**
Open up the freshly baked tailwind.config.js and tell Tailwind where to find your files. Something like this will do:

```
module.exports = {
  content: [
    "./index.html",
    "./src/**/*.{js,ts,jsx,tsx}"
  ],
  theme: {
    extend: {},
  },
  plugins: [],
};
```
(Yes, you need to tell it where to look. Tailwind doesn’t have superpowers, despite how much people rave about it.)

**Step 4: Inject Tailwind’s Magic**
Create a CSS file—let’s call it src/index.css—and add these Tailwind directives:

```
@tailwind base;
@tailwind components;
@tailwind utilities;
```
Then import this file wherever your app starts (probably main.jsx, unless you’ve decided to rewrite the rulebook):
`import './index.css';`

**Step 5: Run the Thing**

Finally, start your development server:
`npm run dev`

Pop open the URL it spits out, and voilà—your Tailwind-powered masterpiece is ready. Well, “masterpiece” might be a stretch, but hey, you’ve got a good foundation.

**Reply by itsbravestone** [reply]
That's amazing way of explaining. Thank you :)
